xmppserver/src/main/java/org/jivesoftware/util/FaviconServlet.java

```diff
@@ -88,15 +92,15 @@ public void init(ServletConfig config) throws ServletException {
             .build();
         // Load the default favicon to use when no favicon was found of a remote host
         try {
-            URL resource = config.getServletContext().getResource("/images/server_16x16.gif");
-            defaultBytes = getImage(resource.toString());
+           defaultBytes = Files.readAllBytes(Paths.get(JiveGlobals.getHomeDirectory(), "plugins/admin/webapp/images/server_16x16.gif"));
         }
-        catch (MalformedURLException e) {
+        catch (final IOException e) {
             LOGGER.warn("Unable to retrieve default favicon", e);
         }
         // Initialize caches.
         missesCache = CacheFactory.createCache("Favicon Misses");
         hitsCache = CacheFactory.createCache("Favicon Hits");
+        sessionManager = SessionManager.getInstance();
```

Comment:
Nit: maybe translate the sessionManager field to a local variable, to avoid keeping a reference? I'm guessing that the servlet is rarely called.